src/cli/rustup_mode.rs

```diff
@@ -862,10 +864,10 @@ async fn update(cfg: &mut Cfg<'_>, opts: UpdateOpts) -> Result<utils::ExitCode>
             }
         }
         if self_update {
-            common::self_update(|| Ok(utils::ExitCode(0)), cfg.process).await?;
+            common::self_update(|| Ok(()), cfg.process).await?;
         }
     } else {
-        common::update_all_channels(cfg, self_update, opts.force).await?;
+        exit_code = exit_code.and(common::update_all_channels(cfg, self_update, opts.force).await?);
```

Comment:
@djc Since `ExitCode::and` essentially simulates `&&` in bash, how about overriding `ops::BitAnd`, and this will become a pretty standard `&=`?

Reply:
Seems okay.